dist_tables/dml.rst

```diff
@@ -38,6 +38,105 @@ For example:
+    url text,
+    view_count bigint,
+    PRIMARY KEY (site_id, day, url)
+  );
+
+  INSERT INTO daily_page_views (day, site_id, url, view_count)
+    -- this is our original select query
+    SELECT view_time::date AS day, site_id, url,
+           count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, site_id, url;
+
+  -- now the results are available right out of the table
+  SELECT day, site_id, url, view_count
+  FROM daily_page_views;
+
+
+Unfortunately as the numbers of tracked web sites and their pages grow, the aggregate query for populating (or as we'll see later, updating) the rollup table slows down. To compensate we can combine rollup tables and distributed computing to scale the application. Since we GROUP BY each site separately we can parallelize the computation across nodes in a distributed database. Each node in a Citus cluster can hold the data for different web sites and the nodes can each compute rollups locally and write a corresponding part of :code:`daily_page_views`. We distribute both :code:`page_views` and :code:`daily_page_views` so that their rows will stay on the same machine when their :code:`site_id` values match.
+
+.. code-block:: postgresql
+
+  -- First distribute the tables. Notice how we're using the
+  -- same distribution column to keep page views and their daily
+  -- summaries on the same machine
+
+  SELECT create_distributed_table('page_views', 'site_id');
+  SELECT create_distributed_table('daily_page_views', 'site_id');
+
+Keeping the tables' information together on each node, i.e. `co-locating <colocation_groups>`_ them, minimizes network traffic between nodes and allows highly parallel execution. In fact for INSERT INTO SELECT to work in Citus, colocation isn't just a good idea, it's the law. Citus requires the source and destination table to be colocated and throws an error if they are not. Citus implements INSERT INTO SELECT by pushing down the select query to each shard. The distributed query execution happens automatically, just use the ordinary SQL command.
+
+.. code-block:: postgresql
+
+  -- Then run the ordinary INSERT INTO SELECT as before
+
+  INSERT INTO daily_page_views (day, site_id, url, view_count)
+    SELECT view_time::date AS day, site_id, url,
+           count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, site_id, url;
+
+
+In summary, INSERT INTO SELECT on Citus requires that:
+
+- The tables queried and inserted are distributed by analogous columns
+- The select query includes the distribution column
+- The insert statement includes the distribution column
+
+Rollups keep statistics queries fast but do require upkeep. New items must be periodically added or existing entries updated. In order that this periodic update be fast we need to do it *incrementally*, meaning without having to re-scan the entire underlying dataset (as, for instance, a materialized view would require). PostgreSQL's upsert feature is what we need.
```

Comment:
@begriffs -- I edited the example query above to roll-up each day (incrementally).

Could we revise these two paragraphs to say that we could also use UPSERT to handle late arriving data?